File: ports/go/src/core/io/bcsdir.go
package io

import (
	"os"
)

// BcsDir abstracts standard directory operations
type BcsDir struct {
	path string
}

func NewBcsDir(path string) *BcsDir {
	return &BcsDir{path: path}
}

// Exists checks if the directory is present on the filesystem
func (d *BcsDir) Exists() bool {
	info, err := os.Stat(d.path)
	if os.IsNotExist(err) {
		return false
	}
	return info.IsDir()
}

// Mkdir creates the directory
func (d *BcsDir) Mkdir() error {
	return os.Mkdir(d.path, 0755)
}

// MkdirPath creates the directory and any necessary parents
func (d *BcsDir) MkdirPath() error {
	return os.MkdirAll(d.path, 0755)
}

File: ports/rust/src/core/io/bcsdir.rs
use std::fs;
use std::path::Path;

// BcsDir abstracts standard directory operations for Rust
pub struct BcsDir {
    path: String,
}

impl BcsDir {
    pub fn new(path: &str) -> Self {
        BcsDir {
            path: path.to_string(),
        }
    }

    pub fn exists(&self) -> bool {
        Path::new(&self.path).is_dir()
    }

    pub fn mkdir(&self) -> std::io::Result<()> {
        fs::create_dir(&self.path)
    }

    pub fn mkdir_path(&self) -> std::io::Result<()> {
        fs::create_dir_all(&self.path)
    }
}
